### DISCUSSION TITLE
Visual Studio Copilot inline suggestions not showing until manually requested, output panel not displaying logs, inline suggestions slow/never appear

### DISCUSSION THREAD

**Discussion by JeremyTCD** [community/community#48470] [Copilot Conversations] [answered]
## Topic Area

Bug

## Versions
Visual Studio 17.4.5
Copilot 1.73.3.0, paid subscription

## Issue 1: Inline suggestions not showing up until manually requested, output panel not displaying logs
Copilot was working fine with "VS + C#" several weeks ago. I took a short break from "VS + C#", returning to it yesterday, at which point I tried to update VS to 17.5.0 and had to roll back to 17.4.5 after Copilot's inline suggestions stopped working entirely. 

Unfortunately, Copilot does not function like before even in 17.4.5.

In the "VS + C#" clip below, after typing the start of a comment, `//`, Copilot fails to automatically generate an inline suggestion. This is not the case with "VSCode + TS", where an inline suggestion is generated automatically. A manual `alt + \` is necessary to trigger inline suggestion generation in "VS + C#".

Also, in the "VS + C#" clip below, the output panel does not display logs. Again, this is not the case with "VSCode + TS", where verbose logs are displayed.

"VS + C#" clip:

https://user-images.githubusercontent.com/11733898/221125844-b97c5ac3-2c31-4f70-a3e8-3f373a4a3d9e.mp4

"VSCode + TS" clip:

https://user-images.githubusercontent.com/11733898/221126926-fa8c5160-6b52-497b-bbdb-9e07884daf39.mp4

## Issue 2: Slow/missing inline suggestions
Additionally in certain cases, spamming `alt + \` does not seem to work or takes a long time to have any effect:

In this "VS + C#" clip I am spamming `alt + \` from the start, the suggestion takes a while to show:

https://user-images.githubusercontent.com/11733898/221130876-7d307cd0-0bac-4cb6-903d-f0a649c7918e.mp4

"VSCode + TS", semantically similar situation to the previous clip, suggestion appears automatically, almost immediately:

https://user-images.githubusercontent.com/11733898/221130897-28c1c452-c2da-492f-b916-55a2dca32c05.mp4

## Conclusion
I understand that parity between languages/IDEs is not guaranteed,  but these are significant regressions, hope they get fixed.

## Edits
Not sure if this is important, I clicked the exclamation mark at the bottom left of the VS window and noticed this: 
![copilot_error](https://user-images.githubusercontent.com/11733898/221158392-ef980432-6cfd-4c10-9331-9c608cba49c5.png)

Just tested Copilot with the same project on a laptop. Versions: VS 17.4.4, Copilot 1.65.0.0. Copilot works as expected in the text editor, though the output panel is empty:

https://user-images.githubusercontent.com/11733898/221171486-3100e782-3c74-49f1-8d6f-6055b8c45793.mp4

**Comment by RobMHarris** [comment]
I'm having exactly the same issue with Visual Studio 2022 (currently on v 17.5, although I'm pretty sure it stopped working before I upgraded to 17.5). It will only give suggestions if I manually select "Trigger Inline Suggestion". I have this problem on two different machines. I've tried reinstalling the extension as well as reinstalling Visual Studio 2022, and disabling ReSharper, but all with no luck.

**Reply by kasthack** [reply]
Yup, same here:

* Microsoft Visual Studio Community 2022 (64-bit) - Preview / Version 17.5.0 Preview 6.0
* Windows 11 Pro / 22H2 / 22621.1265
* No VS extensions.
* VS Code works fine on the same machine / with the same project.

**Reply by leoshusar** [reply]
Same problem. Sometimes it works fine, but sometimes it doesn't or works really dumb.
Recently this happened (inside the comment is what Copilot suggested):
```cs
var newPoints = CalculatePoints(_points);
var newInnerPoints = //_innerPoints;
```
So I tried to help it:
```cs
var newPoints = CalculatePoints(_points);
var newInnerPoints = CalculatePoints(//CalculatePoints(_innerPoints);
```

It felt like it was behind or something.

I too have much better experience in VSC.

**Reply by tilmann1** [reply]
@JeremyTCD @RobMHarris @kasthack @leoshusar

- Are you on VS 17.5 + version 1.73.3.0 of the Copilot VS extension?
- Assuming the above is true, can you please send a screenshot of your IntelliCode settings? (can be found via the menu bar: Tools -> Options -> IntelliCode -> General/Advanced) Please share both the General and Advanced IntelliCode settings, thanks!

**Reply by JeremyTCD** [reply]
> Are you on VS 17.5 + version 1.73.3.0 of the Copilot VS extension?

Yes. Settings:

![intellicode_settings](https://user-images.githubusercontent.com/11733898/221557157-ca4b40b4-7073-449a-a8a4-50d3dcc46f69.png)

Edit: 

My mistake, I'd reverted to 17.4.5. I've updated to 17.5. The issues still persist with these settings:

![intellicode_settings](https://user-images.githubusercontent.com/11733898/221559817-42197a21-9ec3-4e62-b01e-27f83e299cf4.png)

**Reply by RobMHarris** [reply]
Yes VS 17.5 and version 1.73.3.0 of the Copilot VS extension:
![image](https://user-images.githubusercontent.com/8599482/221558694-69badee5-8502-4804-9a35-00e6b770a66f.png)

![image](https://user-images.githubusercontent.com/8599482/221558602-a81ed78b-7b4b-4c74-a5ee-68ec0c12ae62.png)

**Reply by tilmann1** [reply]
@JeremyTCD
Excellent, thank you! Can you please turn on the "Show whole line completions" option in the IntelliCode General settings? That should address the issues that you're seeing.

For reference here are the recommended default settings for IntelliCode that should give you the best Copilot experience:
![Screenshot 2023-02-27 at 12 57 55](https://user-images.githubusercontent.com/104033760/221558397-e28a8041-e621-4f6f-9ff8-e9f65b8f2215.png)
![Screenshot 2023-02-27 at 12 58 19](https://user-images.githubusercontent.com/104033760/221558419-84be4028-a17e-46b4-93ed-e9916c13fb1a.png)

Pleas let me know whether that addressed your issues. Thanks!

**Reply by tilmann1** [reply]
@RobMHarris Please see my response above that should address the problem on your end as well. Thanks!

**Reply by leoshusar** [reply]
I'm on 17.5.0 preview 6 and tried to update extension just before writing that comment.
![image](https://user-images.githubusercontent.com/13891066/221556838-36d72e9f-e0bc-496b-a9c7-a46f4b027b21.png)

Here is my settings, both should be default:
![image](https://user-images.githubusercontent.com/13891066/221557058-e079c1c3-c579-4fbb-a66e-31619987affc.png)

![image](https://user-images.githubusercontent.com/13891066/221557132-1c54be76-8d51-4f4e-95f3-2fb65b571820.png)

I first thought I have to disable VS autocomplete so it doesn't interfere with Copilot here:
![image](https://user-images.githubusercontent.com/13891066/221557455-97391a7e-cf99-40e5-8e84-633a4b37f4d5.png)

but with that I wasn't getting any suggestions at all. So I assumed this needs to be active for Copilot to work.

Unfortunately I lost screenshots with the problem I was writing about above (that's why I wrote it manually), but today I'll be working in VS again, so when (if?) it happens, I'll screenshot and send it.
Would you like anything else to send? Some logs or something?

EDIT: I'm recording a video now :)

**Reply by kasthack** [reply]
@tilmann1 I had my settings matching these suggestions, but VS still has issues with showing Copilot completions while VS Code works flawlessly.

![image](https://user-images.githubusercontent.com/1292818/221560859-3c0204e7-051e-499a-877e-71d4104aaac7.png)

**Reply by RobMHarris** [reply]
> @RobMHarris Please see my response above that should address the problem on your end as well. Thanks!

Yes that's fixed the problem (after a VS restart). Thanks for you help @leoshusar!

**Reply by JeremyTCD** [reply]
@tilmann1 Yeap works. Thanks!

How can I change Copilot output panel verbosity so it works like [this](https://docs.github.com/en/copilot/troubleshooting-github-copilot/troubleshooting-github-copilot-in-visual-studio)?

**Reply by tilmann1** [reply]
@leoshusar If you could update to the stable release of VS 17.5 that would be great 😄

**Reply by leoshusar** [reply]
@tilmann1 I have stable 17.5 on laptop, so I'm going to try it there. I wish switching preview to stable didn't require full reinstall, it's the only reason I'm on preview😄

Anyway here is the video, this is happening on both preview and stable. It's not the "dumbness" I mentioned above, where it was suggesting nonsense (I'll be working on laptop for now so I can see if that happens), it's just not triggering suggestions until I finish a "word" or trigger it manually.

https://user-images.githubusercontent.com/13891066/221575069-a4b32135-bd99-4fc6-9238-63a5c75548a2.mp4

**Reply by leoshusar** [reply]
I didn't need to wait for too long😄
![image](https://user-images.githubusercontent.com/13891066/221577534-8905ee96-c6dc-4557-aa29-b2ef75d33998.png)

I deleted `_usedPolygons.ToList()` from the `ConvertPolygonsToObjectToCut()` method, that's why it's underlined, Copilot suggested the exact method body before I deleted it.

![image](https://user-images.githubusercontent.com/13891066/221578315-81ef902c-20ac-41f4-a88a-1637edecbd12.png)

![image](https://user-images.githubusercontent.com/13891066/221577941-43ee5322-f707-4e20-a3e8-da014d283d16.png)

Here VSC:
![image](https://user-images.githubusercontent.com/13891066/221578425-46968d2c-22df-42cb-98ba-d8c368e0a3be.png)

**Reply by crodriguezSofteng** [reply]
With VS 17.6.2 I don't see the c# deep-learning base model for completions option
![image](https://github.com/community/community/assets/133955229/406f2fda-7b4b-4a09-b70c-40d8fae26d74)

**Comment by tilmann1** [comment, marked as answer]
@JeremyTCD @RobMHarris Great to hear that changing the setting resolved your issues!
@JeremyTCD Regarding log verbosity: You can start VS via the "Developer Command Prompt for VS 2022" app (which comes with the VS install) the following way: "devenv /log" and that will give you quite a verbose output in the GitHub Copilot Output Window. We generally don't recommend this to users because of the large number of messages that you'll see (and the potential performance impact on VS) but it can certainly be helpful when you run into issues to get more insight in what's actually happening.

**Comment by tilmann1** [comment]
@kasthack Given that you're on VS 17.6 Preview 1 with the correct IntelliCode settings this looks like there's a potential regression compared to VS 17.5, we'll investigate further. Note that while we try to fix issues as soon as possible there's no official Copilot support for preview releases. The general recommendation is to use the latest stable VS release 😄

**Comment by tilmann1** [comment]
The latest version of the Copilot VS extension (1.76.0.0) will now show an error when Copilot is disabled via an IntelliCode setting:
![Screenshot 2023-02-28 at 18 45 42](https://user-images.githubusercontent.com/104033760/221936257-34489d86-3fd6-48eb-b907-0bbdfbc0834f.png)

**Comment by it-agrology** [comment]
I have the same problem with copilot which is that I don't get Inline suggestions as I write code. 
I only get suggestions from the Output console.
So, is there a solution that works in order to resolve this issue? 

I have used the above recommended settings for IntelliCode.

![image](https://user-images.githubusercontent.com/108403375/228796020-cb6ff693-ec2d-4a99-882a-3eb870b3d5b3.png)

![image](https://user-images.githubusercontent.com/108403375/228796298-ec61391e-84de-49df-8878-1b69e1d09b67.png)

**Reply by jaekim2172** [reply]
I have the same problem. I am frustrated.

**Reply by ayim** [reply]
@jaekim2172 @it-agrology please shoot me an email at ayim at github dot com if you're still seeing issues.
Would love to screenshare and root cause what's going on with the engineering team.

**Comment by zkxjzmswkwl** [comment]
Installed Copilot chat.

It 

- Changed my syntax colors without asking nor telling me.
- Broke inline completions/suggestions entirely.

I have two extensions installed.

- Github Copilot
- Github Copilot chat

Wonderful to see that VS is now firmly in the category of second-class software in the eyes of Microsoft. This is a product. Something people **pay** for. There's no excuse for the initial user experience to be outright frustrating. 

I had to jump through hoops just to sign in with Github without signing in to Visual Studio itself. (Enabling device flow authorization). The option to "add Github account" only appears if one is already signed in to their Microsoft account within VS. This was comically not accounted for by the developers of both the Github Copilot & Copilot Chat extensions.

But wait! Github Copilot Chat didn't recognize the device flow Github authorization I had just gone through for Github Copilot. So... I have to sign in with Microsoft -> Add Github account.

Voila! Chat works. But wait, Github Copilot is now broken and returning "0 completion results" for even the most trivial instructions, and also Visual Studio Dark theme was changed to an awful iteration of Github's syntax theme and I cannot change it back as it seems to have overridden the Dark theme entirely.

Thanks guys. You're doing a bang-up job.

**Comment by SwissKerim** [comment]
Inline Suggestions also not working for me. Im using Microsoft Visual Studio Professional 2022 (64-bit) - Current Version 17.12.3 -> Fresh Installation
